internal/github/github.go

```diff
@@ -18,66 +18,101 @@ type GitHubUser struct {
 }
 
 func User(username, token string) GitHubUser {
-	// set up the GitHub API endpoint
-	url := fmt.Sprintf("%s/users/%s", BASE_API_URL, username)
-	if len(username) == 0 {
-		log.Println("Loading logged user profile", username, token)
-		url = fmt.Sprintf("%s/user", BASE_API_URL)
+	url := "/user"
+	if len(username) > 0 {
+		url = fmt.Sprintf("/users/%s", username)
 	}
 
 	log.Println("Loading profile", url)
 
-	// make a GET request to the URL
-	client := http.Client{}
-	req, err := http.NewRequest("GET", url, nil)
+	body, err := fetch(url, "GET", token)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalln("Error fetching profile", err)
```

Comment:
I'm still unsure about this.

Sometimes I think internal libs should not break with fatal, we should handle the error on the caller.

But other times I don't see a problem, as the application should fail to proceed in case this doesn't work.


Still not sure what is best... 🤔